dataplane/dplanerc/interface.go

```diff
@@ -582,118 +615,188 @@ func (ni *Reconciler) reconcile(ctx context.Context, config *oc.Interface) {
 			}
 		}
 	}
+}
 
-	if config.GetOrCreateSubinterface(intf.subintf).Enabled != nil {
-		if state.GetOrCreateSubinterface(intf.subintf).Enabled == nil || config.GetSubinterface(intf.subintf).GetEnabled() != state.GetSubinterface(intf.subintf).GetEnabled() {
-			log.V(1).Infof("setting interface %s enabled %t", data.hostifDevName, config.GetSubinterface(intf.subintf).GetEnabled())
-			if data.hostifID != 0 {
-				_, err := ni.hostifClient.SetHostifAttribute(ctx, &saipb.SetHostifAttributeRequest{
-					Oid:        data.hostifID,
-					OperStatus: proto.Bool(config.GetSubinterface(0).GetEnabled()),
-				})
-				if err != nil {
-					log.Warningf("Failed to set oper status of hostif: %v", err)
-				}
-			}
-			if !data.isAggregate {
-				_, err := ni.portClient.SetPortAttribute(ctx, &saipb.SetPortAttributeRequest{
-					Oid:        data.portID,
-					AdminState: proto.Bool(config.GetSubinterface(0).GetEnabled()),
-				})
-				if err != nil {
-					log.Warningf("Failed to set admin state of port: %v", err)
+func (ni *Reconciler) reconcileSubIntf(ctx context.Context, config, state *oc.Interface) {
+	for idx, subintf := range config.Subinterface {
+		intfRef := ocInterface{
+			name:    config.GetName(),
+			subintf: idx,
+		}
+		data, ok := ni.ocInterfaceData[intfRef]
+		// We have a new interface.
+		if !ok {
+			switch {
+			case config.GetAggregation().GetLagType() != oc.IfAggregate_AggregationType_UNSET:
```

Comment:
I think `/interfaces/interface/config/type` tells you this more directly, no? If the type is set to `IETFInterfaces_InterfaceType_ieee8023adLag` then this means that this is a LAG.